### PR TITLE
Complete stubs for email package in python2

### LINKED ISSUE #275: Stubs for parts of email are missing in python 2.7
I found this out when I tried to annotate code containing `from email import message`.

### LINKED ISSUE #685: Missing email.message.Message stub for Python 2
Discovered while trying to use the email.mime.MIMEMultipart stub. As you can see here, the inheritance from email.message.Message is commented out (presumably because the stub doesn't exist yet): https://github.com/python/typeshed/blob/master/stdlib/2/email/mime/base.pyi

Unfortunately, without that stub, the email.mime stubs are not very useful: we're missing `attach`, `as_string`, `__getitem__`, `__setitem__`, etc.:
https://docs.python.org/2/library/email.message.html#email.message.Message

### PR DESCRIPTION
**Pull request by ilinum**
Fix #685, fix #275

### CODE REVIEW

File: stdlib/2/email/base64mime.pyi [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def base64_len(s): ...
```
Comment (JelleZijlstra) [inline stdlib/2/email/base64mime.pyi]: `def base64_len(s: bytes) -> int: ...`

File: stdlib/2/email/iterators.pyi [hunk 2]
```diff
@@ -0,0 +1,3 @@
+def walk(self) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2/email/iterators.pyi]: This is a generator, it returns an Iterator of something or other (Message objects?).

File: stdlib/2/email/iterators.pyi [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+def body_line_iterator(msg, decode: bool = ...) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/2/email/iterators.pyi]: Same for these two.

File: stdlib/2/email/message.pyi [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class Message:
+    preamble = ...
```
Comment (JelleZijlstra) [inline stdlib/2/email/message.pyi]: Looks like `epilogue` is also a public attribute.

File: stdlib/2/email/message.pyi [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    def get_params(self, failobj=..., header=..., unquote: bool = ...): ...
+    def get_param(self, param, failobj=..., header=..., unquote: bool = ...): ...
+    def set_param(self, param, value, header=..., requote: bool = ..., charset=..., language=...) -> None: ...
+    def del_param(self, param, header=..., requote: bool = ...): ...
+    def set_type(self, type, header=..., requote: bool = ...): ...
+    def get_filename(self, failobj=...): ...
+    def get_boundary(self, failobj=...): ...
+    def set_boundary(self, boundary) -> None: ...
+    def get_content_charset(self, failobj=...): ...
+    def get_charsets(self, failobj=...): ...
```
Comment (JelleZijlstra) [inline stdlib/2/email/message.pyi]: Missing the `walk` method (which is added in a weird way: https://github.com/python/cpython/blob/2.7/Lib/email/message.py#L797).

Comment (ilinum) [reply stdlib/2/email/message.pyi]: Wow, good catch!

File: stdlib/2/email/message.pyi (hunk 4 above)
Comment (ilinum) [reply stdlib/2/email/message.pyi]: Yep, missed it!

**Comment by ilinum** [comment]
Code review feedback should be addressed now!

Yes, we can add more types later but this should be a good start.

**Pull request merged**